src/test/java/org/scijava/log/MultiThreadingTest.java

```diff
@@ -0,0 +1,78 @@
+package org.scijava.log;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Check if calling {@link AbstractLogService#channel(String)} always only returns one Logger object per name,
+ * even if multiple Threads try to create the same channels in parallel.
+ *
+ * @author Matthias Arzt
+ */
+public class MultiThreadingTest {
+
+	private LogService logService;
+
+	@Before
+	public void setUp() {
+		logService = new StderrLogService();
+	}
+
+	@Test
+	public void run() throws InterruptedException {
+		List<List<Logger>> results = runParallelCollectResults(10, this::getLoggers);
+		boolean passed = allEqual(results, MultiThreadingTest::referencesInListEqual);
+		assertTrue(passed);
```

Comment:
I am not sure if `assertAllEqual` will be used more than once.
There are many class in `org.scijava.util` what would be more appropriate `ListUtils` or `MiscUtils`?